fillFlush.cpp

```diff
@@ -0,0 +1,92 @@
+#include <algorithm>
+#include <atomic>
+#include <cstdint>
+#include <cstdlib>
+#include <random>
+#include <thread>
+#include <vector>
+
+#include "gflags/gflags.h"
+
+// Flags that control single-threaded behavior
+DEFINE_int32(batch_size, 1000, "Number of pointers owned by a thread at a time");
+DEFINE_int32(batch_thread_migrations, 200, "Number of pointers to free in a run");
+DEFINE_int32(batch_frees, 200, "Number of pointers in a batch to free");
+DEFINE_int32(batch_sleep_ms, 1, "Number of milliseconds to sleep between batch free and alloc");
+
+// Flags that control cross-thread behavior
+DEFINE_int32(num_threads, 1, "Number of threads to run the test");
+DEFINE_int32(shared_buffer_size, 10 * 1000, "Shared buffer size");
+
+// Control parameters
+DEFINE_int32(num_runs, -1, "Number of runs to perform (or -1 to loop forever)");
+DEFINE_int32(malloc_size, 32, "Size of the allocations");
+DEFINE_int32(randseed, 12345, "Random seed, for gesture in the direction of reproducibility");
+
```

Comment:
nit: paste some sample output here for reference?

Reply:
We don't have any output -- this is pretty much purely to stick after `time` or to run `perf` against.

Reply:
Ah I see. Makes sense.